app/assets/javascripts/textbook_transactions.js

```diff
@@ -0,0 +1,428 @@
+// Place all the behaviors and hooks related to the matching controller here.
+// All this logic will automatically be available in application.js.
+// You can use CoffeeScript in this file: http://coffeescript.org/
+
+var listingsData = [],
+	booksData = [];
+
+var ready = function() {
+
+	var listingsToShow,
+		listingList,
+		booksToShow,
+		bookList;
+
+	const default_book_cover = '/assets/icons/no_book.png'
```

Comment:
should actually have rails serve the no_book image if no corresponding image, and not have the javascript be concerned with this - asset pipelining and all that